web/modules/custom/joinup_subscription/src/EventSubscriber/CollectionContentSubscriptionSubscriber.php

```diff
@@ -5,57 +5,16 @@
 namespace Drupal\joinup_subscription\EventSubscriber;
 
 use Drupal\Core\Entity\ContentEntityInterface;
-use Drupal\Core\Entity\EntityPublishedInterface;
-use Drupal\Core\Entity\EntityTypeManagerInterface;
-use Drupal\Core\Logger\LoggerChannelFactoryInterface;
 use Drupal\joinup_group\JoinupGroupHelper;
 use Drupal\joinup_notification\Event\NotificationEvent;
-use Drupal\joinup_notification\JoinupMessageDeliveryInterface;
 use Drupal\joinup_notification\NotificationEvents;
-use Drupal\og\OgMembershipInterface;
+use Drupal\joinup_subscription\DigestFormatter;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 /**
  * Subscriber for compiling content subscription digest messages.
  */
-class CollectionContentSubscriptionSubscriber implements EventSubscriberInterface {
-
-  /**
-   * The entity type manager.
-   *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
-   */
-  protected $entityTypeManager;
-
-  /**
-   * The Joinup message delivery service.
-   *
-   * @var \Drupal\joinup_notification\JoinupMessageDeliveryInterface
-   */
-  protected $messageDelivery;
-
-  /**
-   * The logger channel factory.
-   *
-   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
-   */
-  protected $loggerFactory;
-
-  /**
-   * Constructs a new event subscriber object.
-   *
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
-   *   The entity type manager.
-   * @param \Drupal\joinup_notification\JoinupMessageDeliveryInterface $joinupMessageDelivery
-   *   The Joinup message delivery service.
-   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger
-   *   The logger channel factory.
-   */
-  public function __construct(EntityTypeManagerInterface $entityTypeManager, JoinupMessageDeliveryInterface $joinupMessageDelivery, LoggerChannelFactoryInterface $logger) {
-    $this->entityTypeManager = $entityTypeManager;
-    $this->messageDelivery = $joinupMessageDelivery;
-    $this->loggerFactory = $logger;
-  }
+class CollectionContentSubscriptionSubscriber extends GroupContentSubscriptionSubscriberBase implements EventSubscriberInterface {
```

Comment:
I see no dependency of this module on `collection` module. However, I think this subscriber should live in `collection` module. Then we don't need any inter-module dependencies.

Reply:
I tried it and it does create a dependency to the notification module. I will move it to joinup_notifications where the rest of notifications lie in too.